### ISSUE TITLE
Is it safe to use express-fileupload as a global middleware?

### ISSUE THREAD

**Issue by leafac** [richardgirges/express-fileupload#277]
The [example for express-fileupload](https://github.com/richardgirges/express-fileupload/blob/da968ef0365eba4bad73909737700798d89d2ad7/example/server.js#L9) installs it as a global middleware, but the [documentation for a similar package, Multer](https://github.com/expressjs/multer/tree/6b5fff5feaf740f249b1b2858e5d06009cbd245c#any), says that at least for that package you shouldn’t use the middleware globally. Hence the question in the title.

As far as I understand the issue is that a route that doesn’t expect files could process files by accident and pollute the temporary directory. Is this not an issue with express-fileupload because it stores the files in memory? If so, could it run into this [other issue described in Multer’s `MemoryStorage`](https://github.com/expressjs/multer/tree/6b5fff5feaf740f249b1b2858e5d06009cbd245c#memorystorage)? And how does the [`useTempFile` option](https://github.com/richardgirges/express-fileupload/tree/da968ef0365eba4bad73909737700798d89d2ad7#using-usetempfile-options) affect all this?

Thank you!

**Comment by tanmaymishu** [comment]
I have the same question about memory usage. Let's say the server has 2GB of memory and I have a 2.1 GB FHD video that I need to upload. What would happen in that case?

**Comment by tanmaymishu** [comment]
Looks like the memory issue has been addressed in #89. Using useTempFiles option should solve it:
```js
app.use(fileUpload({
    useTempFiles : true,
    tempFileDir : '/tmp/'
}));
```

**Comment by leafac** [comment]
I guess the conclusion here is:

If you don’t expect to receive huge files, or a burst of small files, then use memory storage and you may use express-fileupload as a global middleware. That’s the simplest approach.

Otherwise, use file storage and only set express-fileupload as a middleware on routes that expect files and know how to clean up after themselves. That’s the slightly more complicated but more robust approach.

I suppose we may close this.

Thanks for the input, @tanmaymishu!
